lib/elixir/lib/enum.ex

```diff
@@ -1716,7 +1716,7 @@ defmodule Enum do
   @spec slice(t, Range.t) :: list
```

Comment:
Leave it as Range.t then since Elixir v1.1 will enforce ranges to be on integers so this will get to work before 1.1 is out.